src/metrics/postgres_metric.ts

```diff
@@ -0,0 +1,74 @@
+import { AbstractMetric, Datasource, MetricId, MetricQuery } from './metric';
+
+import * as _ from 'lodash';
+export class PostgresMetric extends AbstractMetric {
+
+  private _targetName: string; //save first target name, while multi metric not implemented
+
+  constructor(datasource: Datasource, targets: any[], id?: MetricId) {
+    super(datasource, targets, id);
+    this._targetName = targets[0].refId;
+  }
+
+  getQuery(from: number, to: number, limit: number, offset: number): MetricQuery {
+    let queries = this.datasource['data']['queries'];
+    _.forEach(queries, (q) => {
+        q.rawSql = this.processLimitOffset(q.rawSql, limit, offset);
+    });
+    return {
+      url: this.datasource.url,
+      method: 'POST',
+      schema: {
+        data: {
+          from: `${from}`,
+          to: `${to}`,
+          queries: queries
+        }
+      }
+    };
+  }
+
+  getResults(res) {
+    if(res.data === undefined || res.data.results.length < 1) {
+      console.log('datasource return empty response, no data');
+      return {
+        columns: ['timestamp', 'target'],
+        values: []
+      };
+    }
+
+    // TODO: support more than 1 metric (each res.data.results item is a metric)
+    let results = res.data.results[this._targetName];
+    if (results.series === undefined) {
+      return [];
+    }
+
+    let points = results.series[0].points;
+    points.map(p => p.reverse());
```

Comment:
Looks confusing
`.map` doesn't mutate original array
But `.reverse` seem to do
Would be better to do `points.forEach(p => p.reverse());` or `points = points.map(p => p.reverse());`, I think